src/common/application.h

```diff
@@ -52,43 +53,52 @@ class Application
     // Init
     //
 
+    void initialize();
     void trySetConsoleTitle();
     void registerSignalHandlers();
-    void usercheck();
+    void usercheck() const;
     void tryIncreaseRLimits();
-    void tryDisableQuickEditMode();
-    void tryRestoreQuickEditMode();
+    void tryDisableQuickEditMode() const;
+    void tryRestoreQuickEditMode() const;
     void prepareLogging();
 
-    virtual void loadConsoleCommands() = 0;
-
     void markLoaded();
 
+    //
+    // Derived customization
+    //
+
+    virtual auto createEngine() -> std::unique_ptr<Engine> = 0;
+    virtual void registerArguments(Arguments& args) {};
+    virtual void registerCommands(ConsoleService& console) {};
+
     //
     // Runtime
     //
 
-    bool isRunning();
+    bool isRunning() const;
     void requestExit();
 
     // Is expected to block until requestExit() is called and/or isRunning() returns false
-    virtual void run() = 0;
+    void run();
 
-    bool isRunningInCI();
+    bool isRunningInCI() const;
 
     //
     // Member accessors
     //
 
     auto ioContext() -> asio::io_context&;
-    auto args() -> Arguments&;
-    auto console() -> ConsoleService&;
+    auto args() const -> Arguments&;
+    auto console() const -> ConsoleService&;
 
 protected:
     asio::io_context io_context_;
+    asio::signal_set m_signals;
```

Comment:
The naming convention I'm trying to take forward for members is `signals_`, then `signals` for scope-bound things and arguments